libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -169,3 +169,32 @@ smt::Expression dev::solidity::maxValue(IntegerType const& _type)
 {
 	return smt::Expression(_type.maxValue());
 }
+
+void dev::solidity::setSymbolicZeroValue(SymbolicVariable const& _variable, smt::SolverInterface& _interface)
+{
+	setSymbolicZeroValue(_variable.currentValue(), _variable.type(), _interface);
+}
+
+void dev::solidity::setSymbolicZeroValue(smt::Expression _expr, TypePointer const& _type, smt::SolverInterface& _interface)
+{
+	if (isInteger(_type->category()))
+		_interface.addAssertion(_expr == 0);
+	else if (isBool(_type->category()))
+		_interface.addAssertion(_expr == smt::Expression(false));
+}
+
+void dev::solidity::setSymbolicUnknownValue(SymbolicVariable const& _variable, smt::SolverInterface& _interface)
+{
+	setSymbolicUnknownValue(_variable.currentValue(), _variable.type(), _interface);
+}
+
+void dev::solidity::setSymbolicUnknownValue(smt::Expression _expr, TypePointer const& _type, smt::SolverInterface& _interface)
```

Comment:
If we already have an `smt` namespace why don't we use for these public functions?

Reply:
The `smt` namespace has been used in places that are smt-only, e.g. SolverInterface, Sorts. These Symbolic* files still have Solidity stuff in them. We could use the `smt` namespace there too if it makes it better.

Reply:
I'd say the Symbols files are to be used by the SMT subsystem only, there it they belong under `smt`. If it is a lot of changes to move everything, I'd say just move these new functions under `smt` in this PR.

Reply:
Moved just those functions for now.
Created issue #5484 for the rest.